apps/emqx_management/src/emqx_mgmt_data_backup.erl

```diff
@@ -0,0 +1,674 @@
+    {ok, MnesiaBackupName} = do_export_mnesia_tab(TabName, BackupName),
+    NameInArchive = mnesia_backup_name(BackupBaseName, TabName),
+    ok = ?fmt_tar_err(erl_tar:add(TarDescriptor, MnesiaBackupName, NameInArchive, [])),
+    _ = file:delete(MnesiaBackupName),
+    ok.
+
+do_export_mnesia_tab(TabName, BackupName) ->
+    Node = node(),
+    try
+        {ok, TabName, [Node]} = mnesia:activate_checkpoint(
+            [{name, TabName}, {min, [TabName]}, {allow_remote, false}]
+        ),
+        MnesiaBackupName = mnesia_backup_name(BackupName, TabName),
+        ok = filelib:ensure_dir(MnesiaBackupName),
+        ok = mnesia:backup_checkpoint(TabName, MnesiaBackupName),
+        {ok, MnesiaBackupName}
+    after
+        mnesia:deactivate_checkpoint(TabName)
+    end.
+
+-ifdef(TEST).
+tabs_to_backup() ->
+    %% Allow mocking in tests
+    ?MODULE:mnesia_tabs_to_backup().
+-else.
+tabs_to_backup() ->
+    mnesia_tabs_to_backup().
+-endif.
+
+mnesia_tabs_to_backup() ->
+    lists:flatten([M:backup_tables() || M <- find_behaviours(emqx_db_backup)]).
+
+mnesia_backup_name(Path, TabName) ->
+    filename:join([Path, ?BACKUP_MNESIA_DIR, atom_to_list(TabName)]).
+
+is_import_allowed() ->
+    mria_rlog:role() =:= core.
+
+validate_backup(BackupDir, Opts) ->
+    case hocon:files([filename:join(BackupDir, ?META_FILENAME)]) of
+        {ok, #{
+            <<"edition">> := Edition,
+            <<"version">> := Version
+        }} = Meta ->
+            validate(
+                [
+                    fun() -> check_edition(Edition) end,
+                    fun() -> check_version(Version, Opts) end
+                ],
+                Meta
+            );
+        _ ->
+            ?SLOG(error, #{msg => "missing_backup_meta", backup => BackupDir}),
+            {error, missing_backup_meta}
+    end.
+
+validate([ValidatorFun | T], OkRes) ->
+    case ValidatorFun() of
+        ok -> validate(T, OkRes);
+        Err -> Err
+    end;
+validate([], OkRes) ->
+    OkRes.
+
+check_edition(BackupEdition) when BackupEdition =:= <<"ce">>; BackupEdition =:= <<"ee">> ->
+    Edition = bin(emqx_release:edition()),
+    case {BackupEdition, Edition} of
+        {<<"ee">>, <<"ce">>} ->
+            {error, ee_to_ce_backup};
+        _ ->
+            ok
+    end;
+check_edition(BackupEdition) ->
+    ?SLOG(error, #{msg => "invalid_backup_edition", edition => BackupEdition}),
+    {error, invalid_edition}.
+
+check_version(ImportVersion, Opts) ->
+    case parse_version_no_patch(ImportVersion) of
+        {ok, {ImportMajorInt, ImportMinorInt}} ->
+            Version = emqx_release:version(),
+            {ok, {MajorInt, MinorInt}} = parse_version_no_patch(bin(Version)),
+            case ImportMajorInt > MajorInt orelse ImportMinorInt > MinorInt of
+                true ->
+                    %% 4.x backup files are anyway not compatible and will be treated as invalid,
+                    %% before this step.
+                    maybe_print(
```

Comment:
This can be done differently. 
My point was that: it's hard to ensure forward compatibility, but at the same time, disallowing newer version imports may be too strict. 
Alternatively, we can add some info about forward-compatibility to META.hocon, e.g. 
```
min_major_version=5
min_minor_version=1
```
And when we import the backup, we can compare it with EMQX version and disallow import if it's not supported.

Reply:
Group review feedback: disallow 'downgrade' import (if backup is newer than EMQX - don't allow import).

Reply:
Agreed to disallow importing data if backup version is newer than EMQX version.